### PR TITLE
refactor(itinerary): adapt itinerary-page component to upstream

### PR DESCRIPTION
**Pull request by andreashelms**
Apply upstream refactorings to itinerary-page component and sub components

### CODE REVIEW

File: app/component/ItineraryListContainer.js [hunk 1]
```diff
@@ -253,18 +244,15 @@ class ItineraryListContainer extends React.Component {
       bikeAndPublicItinerariesToShow,
       bikeRentAndPublicItinerariesToShow,
       bikeAndParkItinerariesToShow,
-      scooter,
-      parkAndRide,
       car,
-      onDemandTaxi,
```
Comment (hbruch) [inline app/component/ItineraryListContainer.js]: I dont't think we should remove these stadtnavi additions.

File: app/component/ItineraryListContainer.js [hunk 2]
```diff
@@ -308,25 +296,18 @@ class ItineraryListContainer extends React.Component {
             bikeRentAndPublicItinerariesToShow
           }
           bikeAndParkItinerariesToShow={bikeAndParkItinerariesToShow}
-          scooter={scooter}
-          parkAndRide={parkAndRide}
           car={car}
-          onDemandTaxi={onDemandTaxi}
```
Comment (hbruch) [inline app/component/ItineraryListContainer.js]: We should not remove stadtnavi additions

**Review by hbruch — CHANGES_REQUESTED** [review]
Could you please check ItineraryListContainer? I think we should keep parkAndRide, Scooter and OnDemandTaxi.

Rest looks find to me, Thx!

File: app/component/ItineraryListContainer.js (hunk 1 above)
Comment (andreashelms) [reply app/component/ItineraryListContainer.js]: These props were not used in that component at all.

File: app/component/ItineraryListContainer.js (hunk 2 above)
Comment (andreashelms) [reply app/component/ItineraryListContainer.js]: Same here.

**Review by hbruch — APPROVED** [review]
Ok, we checked that scooters results are still provided. For OnDemand, even now herrenberg.stadtnavi.de shows no results, so this is unrelated to this change.

I approve this PR. Thx.

**Pull request merged**